test/test_admin.py

```diff
@@ -879,3 +879,71 @@ def test_add_and_delete_namespaces(admin):
     assert len(db.namespaces()) == 6
 
     db.drop()
+
+
+def test_database_exists_in_databases_list(admin):
+    db = admin.new_database("my_db")
+    all_databases = admin.databases()
+    assert db in all_databases
+    db.drop()
+
+
+def test_datasource_exists_in_datasource_list(admin, music_options):
+    ds = admin.new_datasource("music", music_options)
+    all_datasources = admin.datasources()
+    # try assert, delete on except / finally ?
```

Comment:
If you care about this I'd say you have a fixture that does clean up rather than making the tests do clean up something like this:

```
@pytest.fixture
def integration(admin):
    # any common setup goes here
    yield  # This will run the test
    # or you can do this if you modify the 'admin' fixture your tests can then use `integration` as the admin connection
    # yield admin 

    try:
          # nuke the stardog instance, remove db's, datasources, users, etc
    except Exception:
          pass  
```

That way all the tests you can just assert the thing was done correctly and let the fixture do the cleanup.

Reply:
I added some fixtures for the resources used in these tests. I created another ticket so that we create fixtures for other resources, as I think that should be part of another effort.
https://github.com/stardog-union/pystardog/pull/103